vm/systemSmartContracts/esdt.go

```diff
@@ -612,17 +618,68 @@ func (e *esdt) wipe(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
 	if returnCode != vmcommon.Ok {
 		return returnCode
 	}
+	returnCode = e.wipeTokenFromAddress(args.Arguments[1], token, args.Arguments[0], args.Arguments[0])
+	return returnCode
+}
+
+func (e *esdt) toggleFreezeSingleNFT(args *vmcommon.ContractCallInput, builtInFunc string) vmcommon.ReturnCode {
+	if len(args.Arguments) != 3 {
+		e.eei.AddReturnMessage("invalid number of arguments, wanted 3")
+		return vmcommon.FunctionWrongSignature
+	}
+	token, returnCode := e.basicOwnershipChecks(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+	if !token.CanFreeze {
+		e.eei.AddReturnMessage("cannot freeze")
+		return vmcommon.UserError
+	}
+	if string(token.TokenType) == core.FungibleESDT {
+		e.eei.AddReturnMessage("only non fungible tokens can be freezed per nonce")
+		return vmcommon.UserError
+	}
+
+	if !e.isAddressValid(args.Arguments[2]) {
+		e.eei.AddReturnMessage("invalid address to freeze/unfreeze")
+		return vmcommon.UserError
+	}
+
+	composedArg := append(args.Arguments[0], args.Arguments[1]...)
```

Comment:
which are the desired arguments? Can you please add a comment above the function signature?

Reply:
Is it desired. As this is how we save inside the frozen account